### PR TITLE
update(panel): Add a demo for a menu using the panel.

### LINKED ISSUE #7681: mdPanel: Add a panel menu demo for relative positioning
Setup the menu panel demo. 

Only do this after relative positioning is done.

### PR DESCRIPTION
**Pull request by ErinCoughlan**
Also changes tests to compare actual position values instead of blindly checking for whatever was set in CSS.

Fixes #7681

@ThomasBurleson @jelbourn @KarenParker @gmoothart @DerekLouie Please review.

### CODE REVIEW

File: src/components/panel/demoBasicUsage/script.js [hunk 1]
```diff
@@ -5,6 +5,17 @@ angular.module('panelDemo', ['ngMaterial'])
 
 function BasicDemoCtrl($mdPanel) {
   this._mdPanel = $mdPanel;
+
+  this.desserts = [
+      'Apple Pie',
```
Comment (jelbourn) [inline src/components/panel/demoBasicUsage/script.js]: Too much indent

File: src/components/panel/demoBasicUsage/index.html [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
   <p>
-    Open a panel over the app's content.
+    A panel can be used to create dialogs, menus, and other overlays.
   </p>
 
   <div class="demo-md-panel-content">
     <md-button class="md-primary md-raised" ng-click="ctrl.showDialog($event)">
       Dialog
     </md-button>
+    <md-button class="md-primary md-raised demo-menu-open-button"
```
Comment (jelbourn) [inline src/components/panel/demoBasicUsage/index.html]: This should have some a11y stuff to make it act like a menu for screen readers.

**Comment by ErinCoughlan** [comment]
PTAL @jelbourn I believe I added all the appropriate roles, aria-*, tabindex, and keyboard navigation.  If I missed something, please let me know.

**Comment by jelbourn** [comment]
Seems okay to me, though it would be good to try it out in at least one reader (VoiceOver is probably the easiest). LGTM

**Comment by KarenParker** [comment]
LGTM

**Comment by ErinCoughlan** [comment]
I missed redirecting focus back to the button when the menu closes. I'll do it on `clickOutsideToClose` and `escapeToClose` as well once we have the `onRemovingCallback`.

**Pull request merged**